### ISSUE TITLE
kustomize edit add [label|annotation] should let the shell handle argument separation

### ISSUE THREAD

**Issue by gmolkvk** [kubernetes-sigs/kustomize#2355]
**Goal**
I am trying to inject metadata into a kustomize build using the `kustomize edit add annotation key:value` command. I have several values, and would like to inject them into a single annotation field in JSON format, which is quite common for kubernetes (it also does this for `kubectl.kubernetes.io/last-applied-configuration`)

**Problem**
Reproduce by inserting json like so;

```
$ kustomize version
{Version:kustomize/v3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-11T03:12:59Z GoOs:linux GoArch:amd64}
$ touch kustomization.yaml
$ kustomize edit add annotation "test:{\"foo\":\"bar\",\"hello\":\"world\"}"
``` 

will cause `Error: field: Invalid value: "\"hello\"": name part must consist of alphanumeric characters, [etc..]`

I expect the same will happen with any string value containing a comma

**Root cause**
The current version [does not accept more than one argument](https://github.com/kubernetes-sigs/kustomize/blob/b349351e6ed5d07179af7cc9550a0ddce234af86/kustomize/internal/commands/edit/add/addmetadata.go#L107) and as a consequence [has to interpret arguments by comma separation](https://github.com/kubernetes-sigs/kustomize/blob/b349351e6ed5d07179af7cc9550a0ddce234af86/kustomize/internal/commands/util/util.go#L39).

**Proposed solution**
Let the shell handle argument separation, accepting each key-value pair as a single item in the argument list. No more need to split string by comma.

**Comment by gmolkvk** [comment]
/remove-lifecycle stale

Could someone from the team please review this issue :) Maybe it could be clarified further..?

**Comment by jenting** [comment]
I had the same issue when setting annotation `"helm.sh/hook":pre-install,pre-upgrade`, it would become 
```
commonAnnotations:
  helm.sh/hook: pre-install
  pre-upgrade: ""
```

**Comment by gmolkvk** [comment]
Yes exactly, this confirms my "I expect the same will happen with any string value containing a comma"

**Comment by mbuffa** [comment]
An update on this would be nice :) I think it would improve the tool significantly.

**Comment by Shell32-Natsu** [comment]
fixed in #3064
